### DISCUSSION TITLE
Joint VIMP Question

### DISCUSSION THREAD

**Discussion by BiodiverCD** [kogalur/randomForestSRC#354] [Q&A] [answered]
A straightforward question hopefully:

When calculating joint VIMP using regular permutation importances the manual says that the selected variables are permuted **simultaneously**. However, it's not clear to me whether they are permuted independently of each other (but just at the same time)? Or they are permuted together in a way that preserves the relationships between joint variables?

Many thanks

**Comment by ishwaran** [comment, marked as answer]
They are permuted independently of each other (but at the same time).

**Reply by BiodiverCD** [reply]
Ok great thanks,
Would it ever make sense to do the alternative –i.e. to permute variables as a set and maintain their relationships– as a way to get joint importance of groups that are collinear (for example)? Or would this be interpretation be fine with the current method?

I am thinking of using joint importances as a way to deal with the importance inflation of co-dependent variables. I might even consider dividing the joint importance score by the number of variables in the group, as a mean contribution to the total joint importance. Does that seem vaguely reasonable?

Many thanks for your help and support

**Comment by ishwaran** [comment]
We found permuting the variables independently gave better answers as the variables get "more noised up".  

Ultimately though the method of permutation is never able to preserve the true multivariate distribution whether we are talking about joint VIMP or usual VIMP.  That is the weakness of permutation VIMP.
